src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -330,16 +353,23 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
             return sniHandle;
         }
 
-        private static byte[] MakeMsSqlServerSPN(string fullyQualifiedDomainName, int port = DefaultSqlServerPort)
+        private static byte[] GetSqlServerSPN(string hostNameOrAddress, string instanceName = null)
         {
-            string serverSpn = SqlServerSpnHeader + "/" + fullyQualifiedDomainName + ":" + port;
+            Debug.Assert(!string.IsNullOrWhiteSpace(hostNameOrAddress));
+            IPHostEntry hostEntry = Dns.GetHostEntry(hostNameOrAddress);
+            string fullyQualifiedDomainName = hostEntry.HostName;
+            if (string.IsNullOrWhiteSpace(instanceName))
+            {
+                instanceName = DefaultSqlServerPort.ToString();
+            }
+            string serverSpn = SqlServerSpnHeader + "/" + fullyQualifiedDomainName + ":" + instanceName;
             return Encoding.UTF8.GetBytes(serverSpn);
         }
 
-        private static string GetFullyQualifiedDomainName(string hostNameOrAddress)
+        private static byte[] GetSqlServerSPN(string hostNameOrAddress, int port)
```

Comment:
Consider removing this function and calling the `private static byte[] GetSqlServerSPN(string hostNameOrAddress, string instanceName)` from the caller with the port passed in as a string.